File: src/main/java/com/seungsu/zerock/service/BoardService.java
package com.seungsu.zerock.service;

import com.seungsu.zerock.domain.BoardVO;

import java.util.List;

public interface BoardService {
    void insertBoard(BoardVO board);
    void deleteBoard(Long bno);
    void updateBoard(BoardVO board);
    BoardVO getBoard(Long bno);
    List<BoardVO> getBoardList();
}

File: Week1/src/main/java/com/example/recruitBoard/service/BoardService.kt
package com.example.recruitBoard.service

import com.example.recruitBoard.vo.BoardVO
import com.example.recruitBoard.vo.UserVO


interface BoardService {
    fun getBoardList(): List<BoardVO>
    fun getBoard(seq: Int): BoardVO
    fun insertBoard(vo: BoardVO): Int
    fun deleteBoard(seq: Int): Int
    fun updateBoard(vo: BoardVO): Int
    fun searchBoard(searchKeyword: String? = null): List<BoardVO>
    fun applyBoard(boardVO: BoardVO, userVO: UserVO)
}